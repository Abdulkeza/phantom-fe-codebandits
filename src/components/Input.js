import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';

const Input = ({ name, placeholder, icon, onChange }) => {
  return (
    <div className="flex flex-col">
      <label htmlFor={name} className="capitalize mb-3 font-semibold">
        {name}
      </label>
      <input
        type={name}
        id={name}
        name={name}
        placeholder={placeholder}
        onChange={onChange}
        className="rounded-md mb-5 p-2 border-2 w-full"
      />
      <p className="text-red-500 hidden"></p>
    </div>
  );
};

export default Input;
