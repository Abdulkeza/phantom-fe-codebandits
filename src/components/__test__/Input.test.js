import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../Input.js';

describe('Input', () => {
  it('should render Input', () => {
    const elem = renderer.create(<Input />).toJSON();
    expect(elem).toMatchSnapshot();
  });
});
