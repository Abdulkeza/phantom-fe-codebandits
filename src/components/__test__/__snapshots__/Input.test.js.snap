// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render Input 1`] = `
Array [
  <label
    className="capitalize mb-2 font-semibold"
  />,
  <input
    className="rounded-md mb-2 p-2 border-2 w-full"
  />,
]
`;

exports[`Input should render Input 1`] = `
Array [
  <label
    className="capitalize mb-2 font-semibold"
  />,
  <input
    className="rounded-md mb-2 p-2 border-2 w-full"
  />,
]
`;
